stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -200,18 +200,21 @@ public ProcessingStateMachine(
 
   private void skipRecord() {
     notifySkippedListener(currentRecord);
-    inProcessing = false;
+    markProcessingCompleted();
```

Comment:
❓ @Zelldon why do we need to call this here? inProcessing should never be equal to `true` when we enter the `skipRecord()` method, right? I am asking you because it was already there also for the previous implementation.

Reply:
SkipRecord is called from different places.